src/app/mainview/mainview.component.ts

```diff
@@ -53,9 +53,12 @@ export class MainviewComponent implements OnInit, OnDestroy {
     public selectedRegion;
     private navigationSubscription;
     public selectedRegressionRegion: Array<Regressionregion>;
+    public tempRegressionRegion: Array<Regressionregion>;
```

Comment:
This is a minor thing, but I recommend keeping similar variables consistent, and capitalizing each word (camelCase), so in that case you would name it `tempSelectedRegressionRegion` and the next line `tempSelectedStatisticGroup`